README.md

```diff
@@ -13,66 +13,113 @@ Assumptions:
-| `npm sass`   | dev server w/ scss source maps                      |
-| `npm travis` | executes travis validation script                   |
-
-### Deployment Build Step with DockerFile
-When deploying we use the dockerfile for our build step. We can simulate that process as follows:
-
-```
-# from usaspending-website root, build the container image
-docker build -t usaspending-website .
-# generate static artifacts in ./public
-docker run -i --rm=true -v $(pwd)/public:/node-workspace/public usaspending-website /bin/sh -c 'npm run dev'
-# mount the static artifacts to an nginx image and run
-docker run -p 8020:80 -v $(pwd)/public:/usr/share/nginx/html:ro nginx
-```
-
-The app should now be running at `http://localhost:8020`.
+    ```bash
+    # .env file
+    USAPENDING_API=http://localhost:8000/api
+    # ... next var
+    ```
+    - Confirm Docker Compose picks up vars by running: `docker-compose config` to see the values applied
+    - Also can be applied with `docker run --env-file .env ...`
+    - _NOTE: Default env var values are set in the `Dockerfile` and will be used in `docker run` if none are provided_
+2. Using `export` to set them in the current executing shell environment.
+
+    ```bash
+    $ export USAPENDING_API=http://localhost:8000/api
+    $ npm dev
+    ```
+    - `docker-compose` will also pick these up, but `docker run` will not and fall back to `Dockerfile` defaults.
+3. On the command-line, prior to a shell command, to inject into the sub-shell environment
+
+    ```bash
+    $ USAPENDING_API=http://localhost:8000/api npm dev
+    ```
+
+The latter methods in this list take precedence over environment variables set in an earlier method. 
+- If you build or run `npm` scripts on your local desktop development environment with the `npm` CLI, or switch between this and Docker, setting up your shell with `export` (or a tool like `direnv`) is most flexible. 
+- If you only run scripts/builds in Docker or Docker Compose, using a `.env` file is a simple, clean, config-file-driven approach 
+
+### Scripts
+_Custom and life-cycle scripts to execute, as defined under the `scripts` property in `package.json`_
+
+| Script       | Output                                                      |
+|--------------|-------------------------------------------------------------|
+| `npm start`  | dev server with hot reloading                               |
+| `npm prod`   | generates static assets w/ production optimization          |
+| `npm dev`    | generates static assets w/ development optimization         |
+| `npm lint`   | executes linter                                             |
+| `npm test`   | executes unit tests                                         |
+| `npm sass`   | dev server w/ scss source maps                              |
+| `npm travis` | executes travis validation script                           |
+| `npm ci`     | clean existing Node dependencies and install dependencies   |
+
+### Bundling and Deployment
```

Comment:
```suggestion
### Building the App with Docker
```

Reply:
All docker specific stuff should be pigeon-holed here so not as to confuse the majority of readers who I'm assuming are not using it.

Reply:
I'll take your point about making this a frontend-developer oriented document, and that it is recommended they just run npm and node on their desktop. 

But there are many technical folks (ops people, backend devs) that may want to run the website and not have to go through all the trouble of cluttering their system with node, npm, and node modules. So Docker provides that clean and isolated environment to run this. 

I'll try to tuck it all in that section.